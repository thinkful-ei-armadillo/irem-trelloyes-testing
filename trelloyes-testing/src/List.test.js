import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('<List />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<List 
    cards={[
    {title: "test card1 title", content: "test card1 content"}, 
    {title: "test card2 title", content: "test card2 content"}]} />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List 
        header="Test List Header" 
        cards={[
          {title: "test card1 title", content: "test card1 content"}, 
          {title: "test card2 title", content: "test card2 content"}]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    })
})