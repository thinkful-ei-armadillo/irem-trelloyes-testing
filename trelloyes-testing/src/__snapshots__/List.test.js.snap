// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test List Header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        test card1 title
      </h3>
      <p>
        test card1 content
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        test card2 title
      </h3>
      <p>
        test card2 content
      </p>
    </div>
  </div>
</section>
`;
